File: Lua/calc.lua
while true do
    io.write("Getal 1: ")
    local input = io.read()

    if input == "stop" then
        print("Adios!")
        break
    end

    local getal1 = tonumber(input)

    io.write("Operator (+, -, *, /): ")
    local operator = io.read()

    io.write("Getal 2: ")
    local getal2 = tonumber(io.read())

    if operator == "+" then
        resultaat = getal1 + getal2
    elseif operator == "-" then
        resultaat = getal1 - getal2
    elseif operator == "*" then
        resultaat = getal1 * getal2
    elseif operator == "/" then
        if getal2 == 0 then
                print("Nuh uh you didn't! ")
                break
        else
            resultaat = getal1 / getal2
        end
    else
        io.write("Ongeldige operator")
    end

    print(string.format("%g %s %g = %.2f", getal1, operator, getal2, resultaat))
end

File: Haskell/calc.hs
import Text.Read (readMaybe)
import Text.Printf

bereken :: Double -> String -> Double -> Double
bereken x "+" y = x + y
bereken x "-" y = x - y
bereken x "*" y = x * y
bereken x "/" y = if y /= 0 then x / y else 0
bereken _ _ _   = 0

main :: IO()
main = do
    putStrLn "Getal 1: "
    input1 <- getLine
    
    putStrLn "Operand (+, -, *, /): "
    operand <- getLine
    
    putStrLn "Getal 2: "
    input2 <- getLine
    
    let getal1 = read input1 :: Double
    let getal2 = read input2 :: Double
    
    let resultaat = bereken getal1 operand getal2
    
    printf "%.1f %s %.1f = %.2f\n" getal1 operand getal2 resultaat